### PR TITLE
Marketplace search

### PR DESCRIPTION
**Pull request by kcper**
### All Submissions

* [x] Can you post a screenshot of your changes (if applicable)?

https://user-images.githubusercontent.com/9114200/142836030-2504495c-191b-4150-b15b-4cac56987175.mp4

### New Feature Submissions

* [x] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?
https://github.com/CryptoBlades/marketplace/issues/77
### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

Market searching functionality in the topbard.
It searches skill shop items by name and weapons/characters/shields by id.
I used vue-select for this functionality, had a lot of trouble with adjusting the styles as most of the styles are global and were/are breaking it.
Selecting an item in search bar switches to appropriate tab in the market, but does not filter the results yet.

### CODE REVIEW

**Comment by wintersummermint** [comment]
Reviewing this now and testing

**Comment by wintersummermint** [comment]
Currently not returning any results, might need to rebuild

File: src/App.vue [hunk 1]
```diff
@@ -119,7 +178,37 @@ export default Vue.extend({
...
         })
       },
+
+      watch: {
+        selectedInSearch(newVal: SearchResult) {
+          if(!newVal) {
+            this.searchInput = '';
+            return;
+          } 
+          console.log(newVal.section);
```
Comment (Chriserus) [inline src/App.vue:188]: Unnecessary console log

File: src/store/index.ts [hunk 2]
```diff
@@ -576,6 +576,17 @@ export const store = new Vuex.Store<IState>({
 
         return { seller, nftID, price } as { seller: string, nftID: string, price: string };
     },
+    async fetchSellerOfNft({ state }, { nftContractAddr, tokenId }: { nftContractAddr: string, tokenId: string }) {
+      // getSellerOfNftID
```
Comment (Chriserus) [inline src/store/index.ts:580]: Delete comment

File: src/App.css [hunk 3]
```diff
@@ -454,6 +454,41 @@ body {
...
+
+.search-bar .vs__search::placeholder,
+.search-bar .vs__dropdown-toggle,
+.search-bar .vs__dropdown-menu, 
+.vs__search {
+  min-width: 20em;
+  background: #1e293c;
+  border: none;
+  padding: 4px;
+  color: rgba(255, 255, 255, 0.522);
```
[the remark this answers is not in the document]
Comment (kcper) [reply src/App.css:473]: 0.522 is 50% opacity

**Comment by wintersummermint** [comment]
tested again its working fine now, just rebuilt and reinstall my CB builds and api

**Pull request merged**